tools/tts/tts/tts-api.py

```diff
@@ -0,0 +1,77 @@
+import torch
+from TTS.api import TTS
+import os
+import io
+import gc
+import json
+import subprocess
+import requests
+from flask import Flask, request, send_file, abort
+
+tts = TTS("tts_models/en/vctk/vits", progress_bar=False, gpu=False)
+
+app = Flask(__name__)
+request_count = 0
+
+voice_name_mapping = {}
+
+use_voice_name_mapping = True
+with open("./tts_voices_mapping.json", "r") as file:
+	voice_name_mapping = json.load(file)
+	if len(voice_name_mapping) == 0:
+		use_voice_name_mapping = False
+
+voice_name_mapping_reversed = {v: k for k, v in voice_name_mapping.items()}
+
+authorization_token = "coolio"
+
+@app.route("/tts")
+def text_to_speech():
+	global request_count
+	if authorization_token != request.headers["Authorization"]:
+		abort(401)
+
+	request_count += 1
+	voice = request.args.get("voice", '')
+	if use_voice_name_mapping:
+		voice = voice_name_mapping_reversed[voice]
+	text = request.json.get("text", '')
+
+	filter_complex = request.args.get("filter", '')
+	filter_complex = filter_complex.replace("\"", "")
+
+	result = None
+	with io.BytesIO() as data_bytes:
+		with torch.no_grad():
+			tts.tts_to_file(text=text, speaker=voice, file_path=data_bytes)
+
+		response = requests.get(f"http://ffmpeg-container:5003/ffmpeg?filter={filter_complex}", data=data_bytes.getvalue())
```

Comment:
you have this in the wrong order.

This will still tie up the gpu thread since that happens on this container.

You have to have the ffmpeg container request the tts data from the tts container.